api/services/S3Proxy.js

```diff
@@ -51,8 +51,18 @@ const pipeS3ObjectToResponse = ({ key, res }) => {
 }
 
 const resolveKey = (path) => {
+  if (path.split("/").length <= 4) {
```

Comment:
I have to think way too much about what this means. Can we encode this condition by naming it what we expect it to represent?